### PR TITLE
Merge latest upstream changes into our fork

### PR DESCRIPTION
**Pull request by dougbu**
Hit a few conflicts while doing the `git merge` but important manual bits are in the three later commits. That is, concentrate on 8558fc5, c08b336 and 342e284.

Will follow with small reaction PRs for Testing and Mvc repos. Testing a few other repos but don't expect additional issues.

/cc @BrennanConroy @bricelam

### CODE REVIEW

**Comment by dougbu** [comment]
Can remove the [aspnet/xunit](https://github.com/aspnet/xunit) fork entirely once this goes in. None of the upstream ([xunit/xunit](https://github.com/xunit/xunit)) code builds against DNX anymore.

**Comment by bricelam** [comment]
Any reason not to keep the upstream rename of aspnet -> dnx?

File: NuGet.Config [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <configuration>
   <packageSources>
-    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetvnext/api/v2" />
-    <add key="NuGet.org" value="https://nuget.org/api/v2/" />
+    <add key="AspNetVNext" value="https://www.myget.org/F/aspnetlitedev/api/v2" />
+    <add key="NuGet" value="https://api.nuget.org/v3/index.json" />
```
Comment (bricelam) [inline NuGet.Config]: Should this include the xUnit.net nightly build feed?

**Comment by bricelam** [comment]
:shipit: Should we send a PR upstream for any of the changes we've made in our fork?

File: NuGet.Config (hunk 1 above)
Comment (dougbu) [inline NuGet.Config]: If we included that feed, would also need it in every other `aspnet` repo. Fortunately the runner code works fine without the unreleased xUnit packages.

**Comment by dougbu** [comment]
> Should we send a PR upstream for any of the changes we've made in our fork?

I can't think of much that the upstream branch needs. Now have very few local changes beyond the renames, addition of a few files for our CI, and use of the latest aspnet packages:
```
> git diff upstream/master --find-renames --name-status
M	.gitignore
A	.travis.yml
M	NuGet.Config
M	README.md
A	appveyor.yml
R087	dnx.xunit.sln	aspnet.xunit.sln
A	build.cmd
M	build.sh
M	global.json
A	makefile.shade
R100	src/xunit.runner.dnx/CommandLine.cs	src/xunit.runner.aspnet/CommandLine.cs
R100	src/xunit.runner.dnx/Common/AssemblyExtensions.cs	src/xunit.runner.aspnet/Common/AssemblyExtensions.cs
R100	src/xunit.runner.dnx/Common/DictionaryExtensions.cs	src/xunit.runner.aspnet/Common/DictionaryExtensions.cs
R100	src/xunit.runner.dnx/Common/Guard.cs	src/xunit.runner.aspnet/Common/Guard.cs
R100	src/xunit.runner.dnx/Common/TestDiscoveryVisitor.cs	src/xunit.runner.aspnet/Common/TestDiscoveryVisitor.cs
R100	src/xunit.runner.dnx/Common/XmlTestExecutionVisitor.cs	src/xunit.runner.aspnet/Common/XmlTestExecutionVisitor.cs
R100	src/xunit.runner.dnx/ConsoleRunnerLogger.cs	src/xunit.runner.aspnet/ConsoleRunnerLogger.cs
R100	src/xunit.runner.dnx/DesignTime/DesignTimeExecutionVisitor.cs	src/xunit.runner.aspnet/DesignTime/DesignTimeExecutionVisitor.cs
R100	src/xunit.runner.dnx/DesignTime/DesignTimeTestConverter.cs	src/xunit.runner.aspnet/DesignTime/DesignTimeTestConverter.cs
R100	src/xunit.runner.dnx/DesignTime/SourceInformationProvider.cs	src/xunit.runner.aspnet/DesignTime/SourceInformationProvider.cs
R100	src/xunit.runner.dnx/ParallelismOption.cs	src/xunit.runner.aspnet/ParallelismOption.cs
R100	src/xunit.runner.dnx/Program.cs	src/xunit.runner.aspnet/Program.cs
R100	src/xunit.runner.dnx/Properties/AssemblyInfo.cs	src/xunit.runner.aspnet/Properties/AssemblyInfo.cs
R100	src/xunit.runner.dnx/Properties/GlobalAssemblyInfo.cs	src/xunit.runner.aspnet/Properties/GlobalAssemblyInfo.cs
R100	src/xunit.runner.dnx/Utility/IExecutionVisitor.cs	src/xunit.runner.aspnet/Utility/IExecutionVisitor.cs
R100	src/xunit.runner.dnx/Utility/StackFrameTransformer.cs	src/xunit.runner.aspnet/Utility/StackFrameTransformer.cs
R100	src/xunit.runner.dnx/Utility/Transform.cs	src/xunit.runner.aspnet/Utility/Transform.cs
R100	src/xunit.runner.dnx/Utility/TransformFactory.cs	src/xunit.runner.aspnet/Utility/TransformFactory.cs
R100	src/xunit.runner.dnx/Utility/XmlAggregateVisitor.cs	src/xunit.runner.aspnet/Utility/XmlAggregateVisitor.cs
R100	src/xunit.runner.dnx/Visitors/DiagnosticMessageVisitor.cs	src/xunit.runner.aspnet/Visitors/DiagnosticMessageVisitor.cs
R065	src/xunit.runner.dnx/project.json	src/xunit.runner.aspnet/project.json
A	src/xunit.runner.aspnet/xunit.runner.aspnet.xproj
D	src/xunit.runner.dnx/xunit.runner.dnx.xproj
R100	test/test.xunit.runner.dnx/CommandLineTests.cs	test/test.xunit.runner.aspnet/CommandLineTests.cs
R057	test/test.xunit.runner.dnx/project.json	test/test.xunit.runner.aspnet/project.json
R051	test/test.xunit.runner.dnx/test.xunit.runner.dnx.xproj	test/test.xunit.runner.aspnet/test.xunit.runner.aspnet.xproj
R100	test/test.xunit.runner.dnx/xunit.runner.json	test/test.xunit.runner.aspnet/xunit.runner.json
```

**Comment by dougbu** [comment]
> Any reason not to keep the upstream rename of aspnet -> dnx?

Might be possible to get by with just a distinct package version. But to make that switch, would need to update every aspnet repo. Didn't seem worthwhile at this time. Open to discussion if anyone thinks we should gate this update on matching the upstream package name.

File: NuGet.Config (hunk 1 above)
Comment (bricelam) [inline NuGet.Config]: Good point

**Comment by bricelam** [comment]
> Open to discussion if anyone thinks we should...
The only legitimate reason I can think of is so users don't have to manually update them to dnx on `master` if the want to run tests.

**Comment by dougbu** [comment]
5a12e89

**Pull request closed without merging**